AnnService/inc/Core/Common/WorkSpace.h

```diff
@@ -135,54 +135,71 @@ namespace SPTAG
         };
 
         class DistPriorityQueue {
-            std::unique_ptr<float[]> m_data;
             int m_size;
+            std::unique_ptr<float[]> m_data;
+            int m_length;
             int m_count;
+            
         public:
             DistPriorityQueue() {}
 
             void Resize(int size_) {
-                m_data.reset(new float[size_ + 1]);
-                for (int i = 0; i <= size_; i++) m_data[i] = MaxDist;
                 m_size = size_;
+                m_data.reset(new float[size_ + 1]);
+                
+                m_data[1] = MaxDist;
+                m_length = 1;
                 m_count = size_;
             }
             void clear(int count_) {
```

Comment:
what's the difference of clear() and resize()

Reply:
clear() is used to avoid reallocate when count_ <= m_size.